src/network/network_coordinator.h

```diff
@@ -33,6 +34,12 @@
  *        - Send the client a GC_CONNECT with the peer address.
  *        - a) Client connects, client sends CLIENT_CONNECTED to Game Coordinator.
  *        - b) Client connect fails, client sends CLIENT_CONNECT_FAILED to Game Coordinator.
+ *    2) STUN?
```

Comment:
Maybe a small explanation of what STUN is (supposed) to be, not necessarily here though. Something along the lines (if explained it correctly) would probably suffice:
Both the game server and game client connect to the turn server, after which the turn server tells the game client to connect to the game server's public IP and port that was used to connect to the turn server. This trickery allows two computers that are both behind a NAT to connect to each other without an intermediate server except for setting up the connection.